smartmin/views.py

```diff
@@ -646,14 +648,14 @@ def derive_queryset(self, **kwargs):
         if search_fields and 'search' in self.request.GET:
             terms = self.request.GET['search'].split()
 
-            query = Q(pk__gt=0)
+            term_queries = []
             for term in terms:
-                term_query = Q(pk__lt=0)
+                field_queries = []
                 for field in search_fields:
-                    term_query |= Q(**{field: term})
-                query &= term_query
+                    field_queries.append(Q(**{field: term}))
+                term_queries.append(reduce(operator.or_, field_queries))
 
-            queryset = queryset.filter(query)
+            queryset = queryset.filter(reduce(operator.and_, term_queries))
```

Comment:
Not sure this is better or worse, but for the record `Q()` matches everything and I think you could use `queryset.none()` as the negative case.